rotkehlchen/globaldb/handler.py

```diff
@@ -257,14 +258,33 @@ def add_setting_value(name: str, value: Any) -> None:
     def add_asset(
             asset_id: str,
             asset_type: AssetType,
-            data: Union[EvmToken, dict[str, Any]],
+            data: Union[EvmToken, CustomAsset, dict[str, Any]],
     ) -> None:
         """
         Add an asset in the DB. Either an ethereum token or a user asset.
 
         If it's a user asset the data should be typed. As given in by marshmallow.
 
         May raise InputError in case of error, meaning asset exists or some constraint hit"""
+        if asset_type == AssetType.CUSTOM_ASSET:
+            # TODO: change later to a normal (non-hacky) approach
```

Comment:
You are writing again code that we already have at `import_assets_from_file`

Reply:
You didn't read this comment

Reply:
What exact code do we have there? I don't get